addons/mail/models/statusbar_time_mixin.py

```diff
@@ -0,0 +1,76 @@
+import logging
+from datetime import timedelta
+
+from odoo import fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class StatusbarTimeMixin(models.AbstractModel):
+    _name = "statusbar.time.mixin"
+    _description = "Mixin to compute the time a record spent in each stage"
+    _stage_field = "stage_id"
+
+    json_status_time = fields.Json(
```

Comment:
Maybe something like ``mail_tracking_duration`` ? As json is an implementation detail, and to stay coherent with the namespace ?